### PR TITLE
Events features

### PR DESCRIPTION
**Pull request by MenendezON**
This pull request concerns the implementation of the event of each button in our Calculator app based on React.

### CODE REVIEW

File: src/components/Calculator.js [hunk 1]
```diff
@@ -1,49 +1,33 @@
...
+import Keypad from './keypad';
 
-const Button = ({ name }) => <td>{name}</td>;
+export default class Calculator extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      total: '0',
+      next: null,
+    };
```
Comment (Ibizugbe) [inline src/components/Calculator.js:12]: Just as required in this milestone, we are to make use of `useState` hook to manage our state, `props` to pass data around components, and just as I suggested in the previous comments, you need to convert your class components to functional components to make use of the `useState` hook😉.

image

File: src/components/keypad.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+/* eslint-disable react/prop-types */
+/* eslint-disable react/no-unused-state */
+/* eslint-disable react/destructuring-assignment */
+/* eslint-disable react/prefer-stateless-function */
+import React from 'react';
+import Button from './button';
+
+class Keypad extends React.Component {
+  render() {
+    const items = ['AC', '+/-', '%', '/', '7', '8', '9', 'x', '4', '5', '6', '-', '1', '2', '3', '+', '0', '.', '='];
```
Comment (Ibizugbe) [inline src/components/keypad.js:10]: Currently, when I try to carry out a division operation, an error is displayed which breaks the application, kindly fix this😉.

image

**Review by Ibizugbe — CHANGES_REQUESTED** [review]
Hi @MenendezON 👋🏼👋🏼👋🏼👋🏼, 

require_changes


Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights✅✅✅

- [x] Linters are passing👍🏼.
- [x] Gitflow is used👍🏼.
- [x] the `calculate` and `operate` files are used as required👍🏼.


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by DammyShittu — APPROVED** [review]
Hi @MenendezON,

## APPROVED 🟢

## Highlights
✔️ Project met all technical requirements

Great implementation of all the requirements in this milestone. 👏 

_You have done an amazing job 👍_

Your project is complete!  There is nothing else to say other than... it's time to merge it. :shipit: 

Congratulations! 🎉 

## CHEERS AND HAPPY CODING!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**